plugins/woocommerce/includes/class-wc-download-handler.php

```diff
@@ -654,10 +665,12 @@ private static function download_error( $message, $title = '', $status = 404 ) {
 		 * Since we will now render a message instead of serving a download, we should unwind some of the previously set
 		 * headers.
 		 */
-		header( 'Content-Type: ' . get_option( 'html_type' ) . '; charset=' . get_option( 'blog_charset' ) );
-		header_remove( 'Content-Description;' );
-		header_remove( 'Content-Disposition' );
-		header_remove( 'Content-Transfer-Encoding' );
+		if ( ! headers_sent() ) {
```

Comment:
This check was added because, in the context of running the test suite, I encountered a headers-already-sent error (interestingly, this was because a `system()` call is made during test suite bootstrap and apparently that also triggers an output flush).

Seemed like a safe and straightforward solution, but let me know if any concerns.

Reply:
Perhaps we can refactor this code in a mockable method? But if this is time system then this looks good enough to me. Also let's add a comment here for the reasoning behind this.

Reply:
Yes, although the same thing is at least possible outside of a test scenario—that's just when I encountered this. I expanded it a little to log a specific error in this case (serving the same purpose as a comment but also hopefully helpful in wider terms when merchants or HEs need to troubleshoot this problem).